src/rules/no-unused-definitions.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @fileoverview Rule to prevent unused definitions in Markdown.
+ * @author 루밀LuMir(lumirlumir)
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @import { Definition, FootnoteDefinition } from "mdast";
+ * @import { MarkdownRuleDefinition } from "../types.js";
+ * @typedef {"unusedDefinition" | "unusedFootnoteDefinition"} NoUnusedDefinitionsMessageIds
+ * @typedef {[{ allowDefinitions?: string[], allowFootnoteDefinitions?: string[] }]} NoUnusedDefinitionsOptions
+ * @typedef {MarkdownRuleDefinition<{ RuleOptions: NoUnusedDefinitionsOptions, MessageIds: NoUnusedDefinitionsMessageIds }>} NoUnusedDefinitionsRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoUnusedDefinitionsRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description: "Disallow unused definitions",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/no-unused-definitions.md",
+		},
+
+		messages: {
+			unusedDefinition:
+				"Unexpected unused definition `{{ identifier }}` found.",
+			unusedFootnoteDefinition:
+				"Unexpected unused footnote definition `{{ identifier }}` found.",
+		},
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					allowDefinitions: {
+						type: "array",
+						items: {
+							type: "string",
+						},
+						uniqueItems: true,
+					},
+					allowFootnoteDefinitions: {
+						type: "array",
+						items: {
+							type: "string",
+						},
+						uniqueItems: true,
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+
+		defaultOptions: [
+			{
+				allowDefinitions: ["//"],
```

Comment:
```suggestion
				allowDefinitions: ["//"],
```

`//` is usally used as a definition-style comment.

References:

- https://github.com/eslint/markdown/pull/360#discussion_r2086893446
- https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md053---link-and-image-reference-definitions-should-be-needed
- https://stackoverflow.com/questions/4823468/comments-in-markdown
- https://www.jamestharpe.com/markdown-comments/